web/app/tournament/tourny.py

```diff
@@ -7,87 +7,92 @@
 from app.models import db, Tournament, TournamentPlayer
 from sqlalchemy import update
 
-@tournament.route('/')
+@tournament.route('/', methods=['GET', 'POST'])
 def index():
+    form = TournamentForm()
+    if request.method == 'POST':
+        if form.validate_on_submit():
+            t = Tournament(event_name=form.event_name.data,
+                           venue=form.venue.data,
+                           venue_address=form.venue_address.data,
+                           venue_state=form.venue_state.data,
+                           venue_zip=form.venue_zip.data,
+                           start_date=form.start_date.data,
+                           end_date=form.end_date.data,
+                           director=form.director.data,
+                           director_phone=form.director_phone.data,
+                           director_email=form.director_email.data,
+                           director_address=form.director_address.data,
+                           sponsor=form.sponsor.data,
+                           sponsor_phone=form.sponsor_phone.data,
+                           sponsor_email=form.sponsor_email.data,
+                           sponsor_website=form.sponsor_website.data,
+                           sponsor_address=form.sponsor_address.data,
+                           convener=form.convener.data,
+                           convener_phone=form.convener_phone.data,
+                           convener_email=form.convener_email.data,
+                           convener_website=form.convener_website.data,
+                           convener_address=form.convener_address.data,
+                           pairing=form.pairing.data,
+                           rule_set=form.rule_set.data,
+                           time_controls=form.time_controls.data,
+                           basic_time=form.basic_time.data,
+                           overtime_format=form.overtime_format.data,
+                           overtime_conditions=form.overtime_conditions.data,
+                           komi=form.komi.data,
+                           tie_break1=form.tie_break1.data,
+                           tie_break2=form.tie_break2.data,
+                           submitted=form.submitted.data)
+            db.session.add(t)
+            db.session.commit()
+            # flash("Tournament successfully created.")
+            return redirect(url_for('.index'))
     tournaments = Tournament.query.all()
     return render_template('tournament_index.html',
                     tournaments=tournaments,)
 
-@tournament.route('/<int:tournament_id>/', methods=['GET'])
+@tournament.route('/<int:tournament_id>/', methods=['GET', 'POST', 'DELETE'])
 def show(tournament_id):
     tournament = Tournament.query.get(tournament_id)
     if tournament:
```

Comment:
this pattern can be accomplished with `Tournament.query.get_or_404(game_id)`. It prevents unnecessary indentation of the rest of the method.